contracts/ownership/Whitelist.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.4.18;
+
+
+import "./Ownable.sol";
+
+
+/**
+ * @title Whitelist
+ * @dev The Whitelist contract has a whitelist of addresses, and provides basic authorization control
+ * functions, this simplifies the implementation of "user permissions".
+ */
+contract Whitelist is Ownable {
+  mapping(address => bool) public whitelist;
+  
+  event WhitelistedAddressAdded(address addr);
+  event WhitelistedAddressRemoved(address addr);
+
+  /**
+   * @dev Throws if called by any account that's not whitelisted and not an owner.
+   */
+  modifier onlyWhitelisted() {
+    require(whitelist[msg.sender] || owner == msg.sender);
```

Comment:
The owner can add himself to whitelist if he wants to buy tokens, what about checking only msg.sender?

Reply:
I think this will be more flexible because the owner can add and remove himself if he wants. On the other hand, I can't think of a situation when an owner doesn't want himself to be whitelisted for some operation. I guess it will be more for preventing errors than for authorization in this case, because the owner can whitelist himself at any time?

Reply:
I never saw the owner of the contract receiving tokens. I guess we can leave it but in my experience the owner of the ICO not always receives tokens or ETH, they are forwarded to another wallets, multisig, vestedPayments, etc. That is why O would not assume the owner will want any tokens, and in that case he can whitelist himself.

Reply:
I wouldn't think of this in terms of what the owner wants, I would adjust the behavior to the semantics. This function should only check whether an address is in the whitelist or not as its name promises.

Reply:
Done